### ISSUE TITLE
Error : connection failure with 0.2.0

### ISSUE THREAD

**Issue by Wonderfall** [nvllsvm/Audinaut#7]
Hi, thanks for your work!
I was using your app without any issue until the last 2.0 release. I get no data and when I try a "test connection", I get this message : "Error - Connection failure. A problem occurred communicating with the server. Please check the server address and verify that you can connect using a regular browser on your device.", and yes, I can connect to my Libresonic instance using a regular browser. If it can help, here is my instance : https://music.targaryen.house

Perhaps it's related to okhttp, but I'm not an expert in Android apps to be sure.

**Comment by nvllsvm** [comment]
Definitely related to switching to OkHttp. 

Logcat:
```
03-15 11:51:49.601  2521  2521 W SettingsFragment: javax.net.ssl.SSLHandshakeException: Handshake failed
03-15 11:51:49.601  2521  2521 W SettingsFragment: javax.net.ssl.SSLHandshakeException: Handshake failed
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:429)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at okhttp3.internal.connection.RealConnection.connectTls(RealConnection.java:267)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at okhttp3.internal.connection.RealConnection.establishProtocol(RealConnection.java:237)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at okhttp3.internal.connection.RealConnection.connect(RealConnection.java:148)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at okhttp3.internal.connection.StreamAllocation.findConnection(StreamAllocation.java:186)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at okhttp3.internal.connection.StreamAllocation.findHealthyConnection(StreamAllocation.java:121)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at okhttp3.internal.connection.StreamAllocation.newStream(StreamAllocation.java:100)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:42)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:120)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:179)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at okhttp3.RealCall.execute(RealCall.java:63)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at net.nullsum.audinaut.service.RESTMusicService.ping(RESTMusicService.java:83)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at net.nullsum.audinaut.service.CachedMusicService.ping(CachedMusicService.java:83)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at net.nullsum.audinaut.fragments.SettingsFragment$10.doInBackground(SettingsFragment.java:656)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at net.nullsum.audinaut.fragments.SettingsFragment$10.doInBackground(SettingsFragment.java:644)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at net.nullsum.audinaut.util.BackgroundTask$Task.execute(BackgroundTask.java:201)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at net.nullsum.audinaut.util.BackgroundTask$Task.access$200(BackgroundTask.java:187)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at net.nullsum.audinaut.util.BackgroundTask$TaskRunnable.run(BackgroundTask.java:310)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at java.lang.Thread.run(Thread.java:761)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	Suppressed: javax.net.ssl.SSLHandshakeException: Handshake failed
03-15 11:51:49.601  2521  2521 W SettingsFragment: 		... 28 more
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	Caused by: javax.net.ssl.SSLProtocolException: SSL handshake terminated: ssl=0x7ffedbf43000: Failure in SSL library, usually a protocol error
03-15 11:51:49.601  2521  2521 W SettingsFragment: error:10000410:SSL routines:OPENSSL_internal:SSLV3_ALERT_HANDSHAKE_FAILURE (external/boringssl/src/ssl/s3_pkt.c:610 0x7ffed9e3ab80:0x00000001)
03-15 11:51:49.601  2521  2521 W SettingsFragment: error:1000009a:SSL routines:OPENSSL_internal:HANDSHAKE_FAILURE_ON_CLIENT_HELLO (external/boringssl/src/ssl/s3_clnt.c:764 0x7ffee9cfb70a:0x00000000)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 		at com.android.org.conscrypt.NativeCrypto.SSL_do_handshake(Native Method)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 		at com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:357)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 		... 27 more
03-15 11:51:49.601  2521  2521 W SettingsFragment: Caused by: javax.net.ssl.SSLProtocolException: SSL handshake terminated: ssl=0x7ffedbf43000: Failure in SSL library, usually a protocol error
03-15 11:51:49.601  2521  2521 W SettingsFragment: error:1000042e:SSL routines:OPENSSL_internal:TLSV1_ALERT_PROTOCOL_VERSION (external/boringssl/src/ssl/s3_pkt.c:610 0x7ffed9e3ab80:0x00000001)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at com.android.org.conscrypt.NativeCrypto.SSL_do_handshake(Native Method)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	at com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:357)
03-15 11:51:49.601  2521  2521 W SettingsFragment: 	... 27 more
```

**Comment by nvllsvm** [comment]
All three ciphers `music.targaryen.house` are using are enabled in OkHttp by default:
https://github.com/square/okhttp/blob/master/okhttp/src/main/java/okhttp3/ConnectionSpec.java#L46-L51

``nmap -sT -PN -p 443 music.targaryen.house --script ssl-enum-ciphers.nse``

```
Starting Nmap 7.40 ( https://nmap.org ) at 2017-03-16 08:32 EDT
Nmap scan report for music.targaryen.house (46.4.81.106)
Host is up (0.11s latency).
rDNS record for 46.4.81.106: drogon.targaryen.house
PORT    STATE SERVICE
443/tcp open  https
| ssl-enum-ciphers: 
|   TLSv1.2: 
|     ciphers: 
|       TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384 (secp384r1) - A
|       TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256 (secp384r1) - A
|       TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 (secp384r1) - A
|     compressors: 
|       NULL
|     cipher preference: server
|_  least strength: A

Nmap done: 1 IP address (1 host up) scanned in 6.90 seconds
```

For `sol.nullsum.net`, which is working:

``nmap -sT -PN -p 443 sol.nullsum.net --script ssl-enum-ciphers.nse ``

```Starting Nmap 7.40 ( https://nmap.org ) at 2017-03-16 08:33 EDT
Nmap scan report for sol.nullsum.net (100.11.85.66)
Host is up (0.00086s latency).
rDNS record for 100.11.85.66: pool-100-11-85-66.phlapa.fios.verizon.net
PORT    STATE SERVICE
443/tcp open  https
| ssl-enum-ciphers: 
|   TLSv1.2: 
|     ciphers: 
|       TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 (secp256r1) - A
|       TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384 (secp256r1) - A
|       TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA (secp256r1) - A
|       TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 (secp256r1) - A
|       TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256 (secp256r1) - A
|       TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA (secp256r1) - A
|     compressors: 
|       NULL
|     cipher preference: server
|_  least strength: A

Nmap done: 1 IP address (1 host up) scanned in 0.33 seconds
```

My guess it is that it has something to do with ECDSA.

**Comment by Wonderfall** [comment]
OkHttp 3 should work with TLS_ECDSA ciphers. I'd say maybe HTTP/2 but it seems sol.nullsum.net is also using HTTP/2... For now I have to stick with Audinaut 0.1.2. I'll do further researches on my side. I hope we'll find a solution, I love using your app.

**Comment by gerroon** [comment]
I cant connect to the server as well, Dsub works the browser works.

StatusBarManagerService( 3453): manageDisableList userId=0 what=0x0 pkg=Window{3c4bf2d3 u0 d0 net.nullsum.audinaut/net.nullsum.audinaut.activity.SettingsActivity}
V/WindowManager( 3453): not Base app: Adding window Window{28efcae9 u0 d0 net.nullsum.audinaut/net.nullsum.audinaut.activity.SettingsActivity} at 17 of 25
D/StatusBarManagerService( 3453): manageDisableList userId=0 what=0x0 pkg=Window{28efcae9 u0 d0 net.nullsum.audinaut/net.nullsum.audinaut.activity.SettingsActivity}
D/StatusBarManagerService( 3453): manageDisableList userId=0 what=0x0 pkg=Window{3c4bf2d3 u0 d0 net.nullsum.audinaut/net.nullsum.audinaut.activity.SettingsActivity}
W/SettingsFragment(15372): 	at net.nullsum.audinaut.service.parser.AbstractParser.validate(AbstractParser.java:156)
W/SettingsFragment(15372): 	at net.nullsum.audinaut.service.parser.ErrorParser.parse(ErrorParser.java:47)
W/SettingsFragment(15372): 	at net.nullsum.audinaut.service.RESTMusicService.ping(RESTMusicService.java:84)
W/SettingsFragment(15372): 	at net.nullsum.audinaut.service.CachedMusicService.ping(CachedMusicService.java:83)
W/SettingsFragment(15372): 	at net.nullsum.audinaut.fragments.SettingsFragment$10.doInBackground(SettingsFragment.java:656)
W/SettingsFragment(15372): 	at net.nullsum.audinaut.fragments.SettingsFragment$10.doInBackground(SettingsFragment.java:644)
W/SettingsFragment(15372): 	at net.nullsum.audinaut.util.BackgroundTask$Task.execute(BackgroundTask.java:201)
W/SettingsFragment(15372): 	at net.nullsum.audinaut.util.BackgroundTask$Task.access$200(BackgroundTask.java:187)
W/SettingsFragment(15372): 	at net.nullsum.audinaut.util.BackgroundTask$TaskRunnable.run(BackgroundTask.java:310)
V/WindowManager( 3453): not Base app: Adding window Window{3f68787a u0 d0 net.nullsum.audinaut/net.nullsum.audinaut.activity.SettingsActivity} at 18 of 26
D/StatusBarManagerService( 3453): manageDisableList userId=0 what=0x0 pkg=Window{3f68787a u0 d0 net.nullsum.audinaut/net.nullsum.audinaut.activity.SettingsActivity}
D/StatusBarManagerService( 3453): manageDisableList userId=0 what=0x0 pkg=Window{3f68787a u0 d0 net.nullsum.audinaut/net.nullsum.audinaut.activity.SettingsActivity}
D/StatusBarManagerService( 3453): manageDisableList userId=0 what=0x0 pkg=Window{3c4bf2d3 u0 d0 net.nullsum.audinaut/net.nullsum.audinaut.activity.SettingsActivity}
D/WifiWatchdogStateMachine( 3453): isPackageRunning - top:net.nullsum.audinaut.activity.SettingsActivity

**Comment by edenist** [comment]
Hi, I have just installed Audinaut for the first time, and was getting the exact same problem.

I uninstalled 0.2.0 and installed 0.1.2 and I can now connect without issue.

I'm not particularly well versed with Android development, so I decided to look for any information server-side. 

I'm running libresonic through an Apache proxy, so here is the apache access log for both 0.1.2 and 0.2.0

With 0.2.0:
`[20/Mar/2017:15:15:28 +1100] "GET /rest%2Fping.view?u=test&s=m0cqru0ojpsaik9dvnt5e9joiv&t=b71c1c77e59f8476b4a71da6d311c45d&v=1.2.0&c=Audinaut HTTP/1.1" 404 510 "-" "okhttp/3.6.0"`

With 0.1.2:
`[20/Mar/2017:15:17:53 +1100] "POST /rest/ping.view?u=test&s=dma3chgmom5qk0on62pt63vgbh&t=0ce5c9a094b1706c8ef6c6201362a9fd&v=1.2.0&c=Audinaut HTTP/1.1" 200 523 "-" "Audinaut"`

Main difference I can see is a switch from POST to GET, as well as the forward slash encoding coming through in the GET request. 

Hope this helps.

**Comment by nvllsvm** [comment]
Using the API Level 25 image (with Google APIs), connecting to https://music.targaryen.house and results in the expected incorrect password message.

Using API Level 24 images (with and with Google APIs) results in a failure to connect.

Using a OnePlus 3T running LineageOS 20170320 without Google Apps also results in a failure.

**Comment by nvllsvm** [comment]
Fixed in version [0.2.1](https://github.com/nvllsvm/Audinaut/releases/tag/0.2.1).

**Comment by Wonderfall** [comment]
That's awesome. Thanks!
